<a href="https://colab.research.google.com/github/wiv33/learning-python/blob/master/machine-learning/_000_hello_machine/_001_books/_003_%EB%AA%A8%EB%91%90%EC%9D%98_%EB%94%A5%EB%9F%AC%EB%8B%9D/_000_deeplearning.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [22]:
# machine-learning/_000_hello_machine/_001_books/_003_모두의_딥러닝/_000_deeplearning.ipynb
# 코랩 마운트 인가 후 주석
# from google.colab import drive
# drive.mount('/content/sample_data')
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

import numpy as np
import tensorflow as tf


# 실행할 때마다 같은 결과를 출력하기 위한 설정
np.random.seed(3)
tf.random.set_seed(3)

data_set = np.loadtxt('/content/sample_data/ThoraricSurgery.csv', delimiter=',')

# 환자의 기록과 수술 결과를 X와 Y로 구분하여 저장
X = data_set[:,0:17]
Y = data_set[:,17]

model = Sequential()
model.add(Dense(30, input_dim=17, activation='relu'))
model.add(Dense(1, activation='sigmoid'))

# 딥러닝 실행
model.compile(loss='mean_squared_error', optimizer='adam', metrics=['accuracy'])
model.fit(X, Y, epochs=100, batch_size=10)


Epoch 1/100
47/47 [==============================] - 0s 1ms/step - loss: 0.1485 - accuracy: 0.8426
Epoch 2/100
47/47 [==============================] - 0s 1ms/step - loss: 0.1402 - accuracy: 0.8511
Epoch 3/100
47/47 [==============================] - 0s 1ms/step - loss: 0.1433 - accuracy: 0.8489
Epoch 4/100
47/47 [==============================] - 0s 1ms/step - loss: 0.1396 - accuracy: 0.8511
Epoch 5/100
47/47 [==============================] - 0s 1ms/step - loss: 0.1386 - accuracy: 0.8468
Epoch 6/100
47/47 [==============================] - 0s 1ms/step - loss: 0.1405 - accuracy: 0.8511
Epoch 7/100
47/47 [==============================] - 0s 1ms/step - loss: 0.1337 - accuracy: 0.8489
Epoch 8/100
47/47 [==============================] - 0s 1ms/step - loss: 0.1383 - accuracy: 0.8511
Epoch 9/100
47/47 [==============================] - 0s 1ms/step - loss: 0.1377 - accuracy: 0.8511
Epoch 10/100
47/47 [==============================] - 0s 1ms/step - loss: 0.1327 - accuracy: 0.8489
Epoch 11/

### loss (손실 값) : 예측이 실패할 확률
### accuracy (정확도) : 예측이 성공할 확률

#### 설명
    정확도가 1이면 100% 맞춘다는 의미
    현재 0.8532 이기 때문에 85% 이상 맞춘다는 의미가 됨.

### 중요 포인트
    X 와 Y의 형식과 의미를 파악
    폐암 수술 환자의 생존율 예측하기로,
    
* 진료 기록마다 생존자와 사망자가 있다.
* 해당 진료 기록이 비슷한 경우 군집화로 분류하여 예측하는 방식



